src/main/java/practice/StreamPractice.java

```diff
@@ -75,6 +109,11 @@ public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
      * parametrized with Candidate in CandidateValidator.
      */
     public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        CandidateValidator validator = new CandidateValidator();
+        return candidates.stream()
+                .filter(validator)
+                .map(Candidate::getName)
+                .sorted()
+                .toList();
```

Comment:
```suggestion
        return candidates.stream()
                .filter(CandidateValidator::new)
                .map(Candidate::getName)
                .sorted()
                .toList();
```
Let’s try to simplify it ;)

Reply:
`
return candidates.stream()
                .filter(CandidateValidator::new)
                .map(Candidate::getName)
                .sorted()
                .toList();`
I cannot compile this one, because we are passing constructor reference as predicate